services/src/main/java/org/fao/geonet/api/userfeedback/UserFeedbackAPI.java

```diff
@@ -0,0 +1,424 @@
+@Controller("userfeedback")
+public class UserFeedbackAPI {
+
+    /** The Constant API_PARAM_CSW_SERVICE_IDENTIFIER. */
+    public static final String API_PARAM_CSW_SERVICE_IDENTIFIER = "Service identifier";
+
+    /** The Constant API_PARAM_CSW_SERVICE_DETAILS. */
+    public static final String API_PARAM_CSW_SERVICE_DETAILS = "Service details";
+
+    /**
+     * Delete user feedback.
+     *
+     * @param uuid the uuid
+     * @param request the request
+     * @param response the response
+     * @param httpSession the http session
+     * @return the response entity
+     * @throws Exception the exception
+     */
+    // DELETE
+    @ApiOperation(value = "Removes a user feedback", notes = "Removes a user feedback", nickname = "deleteUserFeedback")
+    @RequestMapping(value = "/userfeedback/{uuid}", produces = MediaType.APPLICATION_JSON_VALUE, method = RequestMethod.DELETE)
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @PreAuthorize("hasRole('Reviewer')")
+    @ApiResponses(value = { @ApiResponse(code = 204, message = "User feedback removed."),
+            @ApiResponse(code = 403, message = ApiParams.API_RESPONSE_NOT_ALLOWED_ONLY_REVIEWER) })
+    @ResponseBody
+    public ResponseEntity deleteUserFeedback(@PathVariable(value = "uuid")
+    final String uuid, final HttpServletRequest request, final HttpServletResponse response, final HttpSession httpSession)
+            throws Exception {
+
+        final ApplicationContext appContext = ApplicationContextHolder.get();
+        final SettingManager settingManager = appContext.getBean(SettingManager.class);
+        final String functionEnabled = settingManager.getValue(Settings.SYSTEM_LOCALRATING_ENABLE);
+
+        if (!functionEnabled.equals(RatingsSetting.ADVANCED)) {
+            return new ResponseEntity(HttpStatus.FORBIDDEN);
+        }
+
+        Log.debug("org.fao.geonet.api.userfeedback.UserFeedback", "deleteUserFeedback");
+
+        final IUserFeedbackService userFeedbackService = getUserFeedbackService();
+
+        userFeedbackService.removeUserFeedback(uuid);
+
+        return new ResponseEntity(HttpStatus.NO_CONTENT);
+    }
+
+    /**
+     * Gets the metadata rating.
+     *
+     * @param metadataUuid the metadata uuid
+     * @param request the request
+     * @param response the response
+     * @param httpSession the http session
+     * @return the metadata rating
+     * @throws Exception the exception
+     */
+    @ApiOperation(value = "Provides an average rating for a metadata record", nickname = "getMetadataUserComments")
+    @RequestMapping(value = "/metadata/{uuid}/userfeedbackrating", produces = MediaType.APPLICATION_JSON_VALUE, method = RequestMethod.GET)
+    @ResponseStatus(value = HttpStatus.OK)
+    @ResponseBody
+    public RatingAverage getMetadataRating(@PathVariable(value = "uuid")
+    final String metadataUuid, final HttpServletRequest request, final HttpServletResponse response, final HttpSession httpSession)
```

Comment:
Add ```@ApiIgnore``` annotation to avoid non API parameters to be displayed in the doc 

```
@ApiIgnore final HttpServletRequest request, @ApiIgnore final HttpServletResponse response, @ApiIgnore final HttpSession httpSession
```

![image](https://user-images.githubusercontent.com/1701393/38668443-40c06820-3e44-11e8-92b7-538abd6ae0b6.png)